modules/http-helper/dummy_server.go

```diff
@@ -28,7 +28,9 @@ func RunDummyServer(t *testing.T, text string) (net.Listener, int) {
 func RunDummyServerE(t *testing.T, text string) (net.Listener, int, error) {
 	port := getNextPort()
 
-	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+	// Create new serve mux so that multiple handlers can be created
+	server := http.NewServeMux()
+	server.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Nice fix. I could've sworn we did this already at some point... But it must've been in some other code base...